google-cloud-firestore/src/test/java/com/google/cloud/firestore/TransactionTest.java

```diff
@@ -959,6 +960,41 @@ public void getShouldThrowWhenInvokedAfterAWriteWithAnAggregateQuery() throws Ex
     assertThat(executionException.getCause()).isInstanceOf(IllegalStateException.class);
   }
 
+  @Test
+  public void givesProperErrorMessageForCommittedTransaction() throws Exception {
+    doReturn(beginResponse())
+        .doReturn(commitResponse(0, 0))
+        .when(firestoreMock)
+        .sendRequest(
+            requestCapture.capture(), ArgumentMatchers.<UnaryCallable<Message, Message>>any());
+    String expectedErrorMessage = "Cannot modify a Transaction that has already been committed.";
+
+    // Commit a transaction.
+    Transaction t = firestoreMock.runTransaction(transaction -> transaction).get();
+
+    // Then run other operations in the same transaction.
+    LocalFirestoreHelper.assertException(
+        () -> {
+          t.set(firestoreMock.collection("foo").document("bar"), map("foo", "bar"));
```

Comment:
Knit: maybe extract the "firestoreMock.collection("foo").document("bar")" into a `docRef` variable.

Reply:
good point. Done.